src/redfish_collector/core/rawCollector.py

```
@@ -49,46 +51,51 @@ def dataJSONWriter(dataRaw,fileDir,fileName, serverAddress):
+    try:
         async with ClientSession(timeout=timeout) as session:
-            # if len(urls) == 1:
-            #     results = await fetch(urls[0],username,password, session)
-            # else:
-            tasks = [fetch(url,username,password, session,serverAddress) for url in urls]
+            tasks = [fetch(url, token, session, serverAddress,asyncio.Semaphore(8)) for url in urls]
             results = await asyncio.gather(*tasks)
             return results
     except Exception as e:
-        logging.error("[%s] There fetch_all error with %s" % (serverAddress,e))
+        logging.error("[%s] Fetch all URL error: %s", serverAddress, e)
```

Ruff (0.14.7)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)